amplitude.js

```diff
@@ -215,42 +221,19 @@ Amplitude.prototype.init = function(apiKey, opt_userId, opt_config, callback) {
       domain: this.options.domain
     });
     this.options.domain = Cookie.options().domain;
+    this._upgradeStoredData();
+    this._loadStoredData();
 
-    _loadCookieData(this);
-
-    this.options.deviceId = (opt_config && opt_config.deviceId !== undefined &&
-        opt_config.deviceId !== null && opt_config.deviceId) ||
-        this.options.deviceId || UUID();
-    this.options.userId = (opt_userId !== undefined && opt_userId !== null && opt_userId) || this.options.userId || null;
-    _saveCookieData(this);
-
-    //log('initialized with apiKey=' + apiKey);
-    //opt_userId !== undefined && opt_userId !== null && log('initialized with userId=' + opt_userId);
-
-    if (this.options.saveEvents) {
-      this._loadSavedUnsentEvents(this.options.unsentKey, '_unsentEvents');
-      this._loadSavedUnsentEvents(this.options.unsentIdentifyKey, '_unsentIdentifys');
-    }
+    this.setUserId(opt_userId || this.options.userId || null);
+    this.setDeviceId(opt_config && opt_config.deviceId !== undefined &&
```

Comment:
better: `this.setDeviceId((opt_config && opt_config.deviceId) || this.options.deviceId || UUID())`